src/library_async.js

```diff
@@ -828,6 +889,155 @@ mergeInto(LibraryManager.library, {
     });
   },
 
+  /*
+   * Layout of an ASYNCIFY Fiber structure
+   *
+   *  0 Asyncify context (Asyncify.currData)
+   *  4 STACKTOP
+   *  8 STACK_MAX
+   * 12 STACK_BASE
+   */
+  $Fibers: {
+    continuations: {},  // fiber pointer -> JS continuation function mapping
+  },
+
+  emscripten_fiber_create__sig: 'iiiii',
+  emscripten_fiber_create__deps: ['$Fibers', 'malloc'],
+  emscripten_fiber_create: function(entryPoint, userData, stack, stackSize) {
+    var fiber = _malloc(16);
```

Comment:
It seems that we are returning heap pointers to C/C++ but only JS knows about the structure so they are compltely opaque?

In which case wouldn't it be faster and simpler to represent the fibre as JS object?  And then have a opaque integer handle returned here? (like just return FibreID++?)

In short, I don't see why this 16 byte structure should live in the heap, and if it was a JS object with three fields there would be far fewer magic numbers here.

Reply:
Yeah, you make a very good point here. Especially considering that the continuation function can only be stored in a JS object anyway.

I think the only reason why I did it this way is because the old coroutines also did it this way. And I don't understand why, either. The structure layout there is also opaque, and different for every backend.